airbyte-cdk/python/unit_tests/sources/declarative/datetime/test_datetime_parser.py

```diff
@@ -44,3 +45,65 @@ def test_format_datetime(test_name, input_dt, datetimeformat, expected_output):
+        ("test_e_datetime_format", "%Y-%m-%e", relativedelta(days=1)),
+        ("test_f_datetime_format", "%Y-%m-%dT%H:%M:%S %f", relativedelta(microseconds=1)),
+        ("test_F_datetime_format", "%F", relativedelta(days=1)),
+        ("test_g_datetime_format", "%g", relativedelta(years=1)),
+        ("test_G_datetime_format", "%G", relativedelta(years=1)),
+        ("test_h_datetime_format", "%Y-%h", relativedelta(months=1)),
+        ("test_H_datetime_format", "%Y-%m-%dT%H", relativedelta(hours=1)),
+        ("test_I_datetime_format", "%Y-%m-%d %I %p", relativedelta(hours=1)),
+        ("test_j_datetime_format", "%Y %j", relativedelta(days=1)),
+        ("test_k_datetime_format", "%Y-%m-%dT%k", relativedelta(hours=1)),
+        ("test_l_datetime_format", "%Y-%m-%d %l %p", relativedelta(hours=1)),
+        ("test_m_datetime_format", "%Y-%m", relativedelta(months=1)),
+        ("test_M_datetime_format", "%Y-%m-%dT%H:%M", relativedelta(minutes=1)),
+        ("test_r_datetime_format", "%Y-%m-%d %r", relativedelta(seconds=1)),
+        ("test_R_datetime_format", "%Y-%m-%d %R", relativedelta(minutes=1)),
+        ("test_s_datetime_format", "%s", relativedelta(seconds=1)),
+        ("test_S_datetime_format", "%Y-%m-%dT%H:%M:%S", relativedelta(seconds=1)),
+        ("test_T_datetime_format", "%Y-%m-%d %T", relativedelta(seconds=1)),
+        ("test_u_datetime_format", "%Y %W %u", relativedelta(days=1)),
+        ("test_U_datetime_format", "%Y %U", relativedelta(weeks=1)),
+        ("test_V_datetime_format", "%Y %V", relativedelta(weeks=1)),
+        ("test_w_datetime_format", "%Y %W %w", relativedelta(days=1)),
+        ("test_W_datetime_format", "%Y %W", relativedelta(weeks=1)),
+        ("test_x_datetime_format", "%x", relativedelta(days=1)),
+        ("test_X_datetime_format", "%Y-%m-%d %X", relativedelta(seconds=1)),
+        ("test_y_datetime_format", "%C%y", relativedelta(years=1)),
+        ("test_Y_datetime_format", "%Y", relativedelta(years=1)),
+        ("test_+_datetime_format", "%+", relativedelta(seconds=1)),
+    ],
+)
+def test_find_most_granular_timedelta(test_name, datetime_format, expected_output):
+    parser = DatetimeParser()
+    output_delta = parser.find_most_granular_timedelta(datetime_format)
+    assert output_delta == expected_output
+
+
+def test_given_most_granular_unit_is_not_parametrized_using_directive_when_find_most_granular_timedelta_then_return_less_granular_delta():
+    output_delta = DatetimeParser().find_most_granular_timedelta("%Y-%m-%dT%H:00:00")
+    _relativedelta_is_bigger_than(output_delta, datetime.timedelta(seconds=1))
+
+
+def test_given_no_known_directives_when_find_most_granular_timedelta_then_raise_error():
+    with pytest.raises(ValueError):
+        DatetimeParser().find_most_granular_timedelta("there is no known directive in this string %q")
+
+
+def _relativedelta_is_bigger_than(relative_delta: relativedelta, time_delta: datetime.timedelta):
```

Comment:
`relativedelta` does not support `__lt__` because there are not ways to know if `relativedelta(days=29) > relativedelta(months=1)`. Therefore, I found a way to evaluate this and hid it in a method